bundle/Controller/Admin/AdminController.php

```diff
@@ -271,10 +293,87 @@ public function handleCollectionAction(Request $request): RedirectResponse
         return $this->redirectToRoute('netgen_information_collection.route.admin.view', ['collectionId' => $collectionId]);
     }
 
+    /**
+     * @throws NonUniqueResultException
+     */
+    public function editAction(Request $request, int $collectionId): RedirectResponse|Response
+    {
+        $this->checkEditPermissions();
+
+        $collection = $this->service->getCollection(new CollectionId($collectionId));
+
+        $location = $collection->getContent()->contentInfo->getMainLocation();
+
+        $form = $this->builder->createUpdateFormForLocation($location, $collection)->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var InformationCollectionStruct $struct */
+            $struct = $form->getData()->payload;
+
+            /** @var ContentType $contentType */
+            $contentType = $form->getData()->definition;
+
+            $ezInfoCollection = $this->infoCollectionRepository->find($collectionId);
+
+            if ($ezInfoCollection === null) {
+                throw new NotFoundHttpException();
+            }
+
+            $ezInfoCollection->setModified(time());
+
+            $this->infoCollectionRepository->save($ezInfoCollection);
+
+            foreach ($struct->getCollectedFields() as $fieldDefIdentifier => $value) {
+                if ($value === null) {
+                    continue;
+                }
+
+                $fieldDefinition = $contentType->getFieldDefinition($fieldDefIdentifier);
+
+                $legacyValue = $this->factory->getLegacyValue($value, $fieldDefinition);
```

Comment:
This does not seem to be correct, `$value` is an instance of `\Ibexa\Contracts\ContentForms\Data\Content\FieldData`, and the method expects an instance of `\Ibexa\Core\FieldType\Value`.

Reply:
Fixed in 4dc434f3 commit

Reply:
I meant on the line above, `$struct->getCollectedFields()` returns an array of `\Ibexa\Contracts\ContentForms\Data\Content\FieldData` instances, and `$value` is in facet and instance of `FieldData`. If `getLegacyValue()` accepts an instance of `\Ibexa\Contracts\Core\FieldType\Value`, how does that even work?

Reply:
I think the inline type hint above is actually wrong.

Reply:
@pspanja fixed in 89702d52 commit